### PR TITLE
NFDIV-790: Event and state changes from Submitted state

### PR DESCRIPTION
**Pull request by SomeshRA**
### JIRA link (if applicable) ###
NFDIV-790

### Change description ###
Events and state changes from Submitted state

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ x] No
```

### CODE REVIEW

File: src/test/java/uk/gov/hmcts/divorce/caseworker/event/IssueApplicationTest.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    @InjectMocks
+    private IssueApplication issueApplication;
+
+    @Test
+    void shouldAddConfigurationToConfigBuilder() {
+        final Set<State> stateSet = Set.of(State.class.getEnumConstants());
+        final ConfigBuilderImpl<CaseData, State, UserRole> configBuilder = new ConfigBuilderImpl<>(CaseData.class, stateSet);
+
+        issueApplication.configure(configBuilder);
+        assertThat(configBuilder.getEvents().get(0).getId(), is(ISSUE_APPLICATION));
```
Comment (nitinprabhu) [inline src/test/java/uk/gov/hmcts/divorce/caseworker/event/IssueApplicationTest.java:30]: please use assertj for consistency

File: src/test/java/uk/gov/hmcts/divorce/caseworker/event/RefundFromSubmittedTest.java [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    @InjectMocks
+    private RefundFromSubmitted refundFromSubmitted;
+
+    @Test
+    void shouldAddConfigurationToConfigBuilder() {
+        final Set<State> stateSet = Set.of(State.class.getEnumConstants());
+        final ConfigBuilderImpl<CaseData, State, UserRole> configBuilder = new ConfigBuilderImpl<>(CaseData.class, stateSet);
+
+        refundFromSubmitted.configure(configBuilder);
+        assertThat(configBuilder.getEvents().get(0).getId(), is(REFUND));
```
Comment (nitinprabhu) [inline src/test/java/uk/gov/hmcts/divorce/caseworker/event/RefundFromSubmittedTest.java:29]: same here

File: src/main/java/uk/gov/hmcts/divorce/caseworker/event/CaseworkerAwaitingDocuments.java [hunk 3]
```diff
@@ -23,14 +23,14 @@
 @Component
 public class CaseworkerAwaitingDocuments implements CCDConfig<CaseData, State, UserRole> {
 
-    public static final String CASEWORKER_AWAITING_DOCUMENTS = "caseworker-awaiting-documents";
+    public static final String CASEWORKER_AWAITING_DOCUMENTS = "awaitingDocuments";
```
Comment (linusnorton) [inline src/main/java/uk/gov/hmcts/divorce/caseworker/event/CaseworkerAwaitingDocuments.java:26]: I would leave this as is

File: src/main/java/uk/gov/hmcts/divorce/caseworker/event/CaseworkerAwaitingDocuments.java [hunk 4]
```diff
@@ -23,14 +23,14 @@
...
 
     @Override
     public void configure(final ConfigBuilder<CaseData, State, UserRole> configBuilder) {
         new PageBuilder(configBuilder
             .event(CASEWORKER_AWAITING_DOCUMENTS)
             .forStateTransition(Submitted, AwaitingDocuments)
-            .name("Awaiting documents")
+            .name("Awaiting Applicant")
```
Comment (linusnorton) [inline src/main/java/uk/gov/hmcts/divorce/caseworker/event/CaseworkerAwaitingDocuments.java:33]: I would also leave this as is

File: src/main/java/uk/gov/hmcts/divorce/caseworker/event/IssueApplication.java [hunk 5]
```diff
@@ -22,18 +22,21 @@
 
 @Component
 public class IssueApplication implements CCDConfig<CaseData, State, UserRole> {
-    public static final String ISSUE_APPLICATION = "issue-application";
+    public static final String ISSUE_APPLICATION = "issueFromSubmitted";
```
Comment (linusnorton) [inline src/main/java/uk/gov/hmcts/divorce/caseworker/event/IssueApplication.java:25]: caseworker-issue-application

File: src/main/java/uk/gov/hmcts/divorce/caseworker/event/RefundFromSubmitted.java [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+import static uk.gov.hmcts.divorce.common.model.UserRole.CASEWORKER_DIVORCE_COURTADMIN_BETA;
+import static uk.gov.hmcts.divorce.common.model.UserRole.CASEWORKER_DIVORCE_COURTADMIN_LA;
+import static uk.gov.hmcts.divorce.common.model.UserRole.CASEWORKER_DIVORCE_SOLICITOR;
+import static uk.gov.hmcts.divorce.common.model.UserRole.CASEWORKER_DIVORCE_SUPERUSER;
+import static uk.gov.hmcts.divorce.common.model.access.Permissions.CREATE_READ_UPDATE;
+import static uk.gov.hmcts.divorce.common.model.access.Permissions.READ;
+
+@Component
+public class RefundFromSubmitted implements CCDConfig<CaseData, State, UserRole> {
+    public static final String REFUND = "refundFromSubmitted";
```
Comment (linusnorton) [inline src/main/java/uk/gov/hmcts/divorce/caseworker/event/RefundFromSubmitted.java:22]: caseworker-refund

File: src/main/java/uk/gov/hmcts/divorce/caseworker/event/RefundFromSubmitted.java [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+
+    @Override
+    public void configure(final ConfigBuilder<CaseData, State, UserRole> configBuilder) {
+        new PageBuilder(configBuilder
+            .event(REFUND)
+            .forStateTransition(Submitted,Submitted)
```
Comment (linusnorton) [inline src/main/java/uk/gov/hmcts/divorce/caseworker/event/RefundFromSubmitted.java:28]: Shouldn't this be Submitted -> Refunded or something like that? Or does it just stay as Submitted?

File: src/main/java/uk/gov/hmcts/divorce/caseworker/event/CaseworkerAwaitingDocuments.java (hunk 3 above)
Comment (SomeshRA) [reply src/main/java/uk/gov/hmcts/divorce/caseworker/event/CaseworkerAwaitingDocuments.java:26]: Reverted.

File: src/main/java/uk/gov/hmcts/divorce/caseworker/event/RefundFromSubmitted.java (hunk 7 above)
Comment (SomeshRA) [reply src/main/java/uk/gov/hmcts/divorce/caseworker/event/RefundFromSubmitted.java:28]: As per AC5 : Event Refund would have this state changes Submitted -> Refunded -> Submitted

File: src/test/java/uk/gov/hmcts/divorce/caseworker/event/IssueApplicationTest.java (hunk 1 above)
Comment (SomeshRA) [reply src/test/java/uk/gov/hmcts/divorce/caseworker/event/IssueApplicationTest.java:30]: Reverted to assertj

File: src/test/java/uk/gov/hmcts/divorce/caseworker/event/RefundFromSubmittedTest.java (hunk 2 above)
Comment (SomeshRA) [reply src/test/java/uk/gov/hmcts/divorce/caseworker/event/RefundFromSubmittedTest.java:29]: Reverted to assertj

File: src/main/java/uk/gov/hmcts/divorce/caseworker/event/RefundFromSubmitted.java [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+            .displayOrder(5)
+            .showSummary()
+            .explicitGrants()
+            .grant(CREATE_READ_UPDATE,
+                CASEWORKER_DIVORCE_COURTADMIN_BETA,
+                CASEWORKER_DIVORCE_COURTADMIN)
+            .grant(READ,
+                CASEWORKER_DIVORCE_SOLICITOR,
+                CASEWORKER_DIVORCE_SUPERUSER,
+                CASEWORKER_DIVORCE_COURTADMIN_BETA,
```
[the remark this answers is not in the document]
Comment (SomeshRA) [reply src/main/java/uk/gov/hmcts/divorce/caseworker/event/RefundFromSubmitted.java]: Removed it from READ grant.

**Review by linusnorton — APPROVED** [review]

**Pull request merged**